static/app/components/globalAppStoreConnectUpdateAlert/updateAlert.tsx

```diff
@@ -1,67 +1,277 @@
-import {useContext} from 'react';
+import {Fragment, useContext, useEffect, useState} from 'react';
 import styled from '@emotion/styled';
+import find from 'lodash/find';
+import mapValues from 'lodash/mapValues';
 
+import {Client} from 'app/api';
 import Alert from 'app/components/alert';
+import Button from 'app/components/button';
+import Link from 'app/components/links/link';
 import AppStoreConnectContext from 'app/components/projects/appStoreConnectContext';
-import {IconRefresh} from 'app/icons';
+import {IconClose, IconRefresh} from 'app/icons';
+import {t} from 'app/locale';
 import space from 'app/styles/space';
 import {Organization, Project} from 'app/types';
+import {AppStoreConnectStatusData} from 'app/types/debugFiles';
+import {promptIsDismissed} from 'app/utils/promptIsDismissed';
+import withApi from 'app/utils/withApi';
+
+const APP_STORE_CONNECT_UPDATES = 'app_store_connect_updates';
 
 type Props = {
+  api: Client;
   organization: Organization;
   project?: Project;
   Wrapper?: React.ComponentType;
   isCompact?: boolean;
   className?: string;
 };
 
-function UpdateAlert({Wrapper, project, className}: Props) {
+function UpdateAlert({api, Wrapper, isCompact, project, organization, className}: Props) {
   const appStoreConnectContext = useContext(AppStoreConnectContext);
+  const [isDismissed, setIsDismissed] = useState(false);
+  const [firstError, setFirstError] = useState(
+    find(appStoreConnectContext, source => source.credentials.status === 'invalid')
```

Comment:
you can use js ` .find` here 😉 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find